web/vtadmin/src/components/routes/workflow/WorkflowDetails.tsx

```diff
@@ -0,0 +1,260 @@
+import { vtctldata } from '../../../proto/vtadmin';
+import { DataCell } from '../../dataTable/DataCell';
+import { StreamStatePip } from '../../pips/StreamStatePip';
+import { ThrottleThresholdSeconds } from '../Workflows';
+
+interface Props {
+    clusterID: string;
+    keyspace: string;
+    name: string;
+}
+
+const SUMMARY_COLUMNS = ['Stream Status', 'Traffic Status', 'Max VReplication Lag', 'Reverse Workflow'];
+
+const LOG_COLUMNS = ['Type', 'State', 'Updated At', 'Message', 'Count'];
+
+const TABLE_COPY_STATE_COLUMNS = ['Table Name', 'Total Bytes', 'Bytes Copied', 'Total Rows', 'Rows Copied'];
+
+const STREAM_COLUMNS = ['Stream', 'State', 'Message', 'Transaction Timestamp', 'Database Name'];
+
+export const WorkflowDetails = ({ clusterID, keyspace, name }: Props) => {
+    const { data: workflowData } = useWorkflow({ clusterID, keyspace, name });
+
+    const { data: workflowsData = [] } = useWorkflows();
+
+    const { data: workflowStatus } = useWorkflowStatus({
+        clusterID,
+        keyspace,
+        name,
+    });
+
+    const reverseWorkflow = getReverseWorkflow(workflowsData, workflowData);
+
+    const tableCopyStates = getTableCopyStates(workflowStatus);
+
+    const streams = useMemo(() => {
+        const rows = getStreams(workflowData).map((stream) => ({
+            key: formatStreamKey(stream),
+            ...stream,
+        }));
+
+        return orderBy(rows, 'streamKey');
+    }, [workflowData]);
+
+    const getStreamsSummary = (streamList: typeof streams): string => {
+        const numStreamsByState: { [key: string]: number } = {
+            Copying: 0,
+            Throttled: 0,
+            Stopped: 0,
+            Running: 0,
+            Error: 0,
+        };
+        streamList.forEach((stream) => {
+            var isThrottled =
+                Number(stream.throttler_status?.time_throttled?.seconds) > Date.now() / 1000 - ThrottleThresholdSeconds;
+            const streamState = isThrottled ? 'Throttled' : stream.state;
+            if (streamState) {
+                numStreamsByState[streamState]++;
+            }
+        });
+        const states = Object.keys(numStreamsByState);
+        let message = '';
+        states.forEach((state) => {
+            if (numStreamsByState[state]) {
+                let desc = state;
+                if (state === 'Error') {
+                    desc = 'Failed';
+                }
+                desc += numStreamsByState[state] > 1 ? ' Streams' : ' Stream';
+                message += `${numStreamsByState[state]} ${desc}. `;
+            }
+        });
+        return message;
+    };
+
+    const workflowSummary = {
+        streamSummary: getStreamsSummary(streams),
+        workflowStatus,
+        workflowData,
+        reverseWorkflow,
+    };
+
+    const renderSummaryRows = (rows: (typeof workflowSummary)[]) => {
+        return rows.map((row) => {
+            const reverseWorkflow = row.reverseWorkflow;
+            return (
+                <tr>
```

Comment:
You'll want to supply a key here, since it's a mapped react component:
```suggestion
                <tr key={row.workflowData.workflow.name}>
```
or another unique identifier!

Reply:
Done.

Reply:
Awesome, thank you!